lib/krikri/harvester.rb

```diff
@@ -100,9 +100,9 @@ def get_record(_)
     # This should be idempotent so it can be safely retried on errors.
     #
     # @return [Boolean]
-    def run
+    def run(activity_uri = nil)
       log :info, 'harvest is running'
-      records.each(&:save)
+      records.each { |rec| rec.save(activity_uri) }
```

Comment:
Maybe I'm asking this in the wrong context, but does this allow us to run partial harvests?